src/backend/distributed/utils/colocation_utils.c

```diff
@@ -660,6 +663,19 @@ UpdateRelationColocationGroup(Oid distributedRelationId, uint32 colocationId)
 
 	systable_endscan(scanDescriptor);
 	heap_close(pgDistPartition, NoLock);
+
+	shouldSyncMetadata = ShouldSyncTableMetadata(distributedRelationId);
+	if (shouldSyncMetadata)
+	{
+		StringInfo command = makeStringInfo();
+		char *relationName = get_rel_name(distributedRelationId);
+		appendStringInfo(command, "UPDATE pg_dist_partition "
+								  "SET colocationid='%d' "
+								  "WHERE logicalrelid='%s'::regclass",
```

Comment:
I have tried to change this by looking into how they are done in other functions of `metadata_sync.c`